### PR TITLE
Fix/retry beagle error

### PR DESCRIPTION
**Pull request by Bleil**
**- What I did**
Following the pattern of the component on the Android environment, I created the same behavior on the component, allowing the user to retry the action that he was trying to do. Also, I added the errors list, that is provided by the core, showing to the user what could happened.

P.S.: The detailed messages are hidden and only will be displayed if the user clicks on "Show details".

**- How I did it**
Followed the pattern of components already developed on the project, keeping the behavior of Android environment and added useful information about what happened, giving details about the errors.

**- How to verify it**
1. After building the project, link it using yarn and create a new React project installing beagle. 
2. Then link your built project
3. On your test React project, create a JSON with this content:
`{
            "_beagleComponent_": "beagle:lazycomponent",
            "path": "/listview.json",
            "initialState": {
                "_beagleComponent_": "beagle:text",
                "text": "Loading the screen, please wait",
                "textColor": "grey",
                "alignment": "CENTER"
            }
        }`
4.  Then start your application, it should display the error message, because we do not have created the "listview.json" that _beagle:lazycomponent_ requires, showing you the error component.

**- Description for the changelog**

### CODE REVIEW

File: packages/beagle-react/package.json [hunk 1]
```diff
@@ -24,7 +24,9 @@
     "unlink-beagle": "cd dist && yarn unlink && cd ../node_modules/react && yarn unlink && cd ../react-dom && yarn unlink && cd ../.."
   },
   "dependencies": {
+    "@types/react-native": "^0.63.46",
```
Comment (Tiagoperes) [inline packages/beagle-react/package.json:27]: pq essas dependências do RN?

File: packages/beagle-react/src/components/BeagleError/index.tsx [hunk 2]
```diff
@@ -1,31 +1,76 @@
...
   return (
-    <div className={className}>
+    <StyledRetryContainer className={className}>
       <BeagleText text="Sorry!" textColor="red" />
       <BeagleText text="An unexpected error happened while loading your page." />
-    </div>
+      <StyledRetryButton
+        type="button"
+        data-testid="retry"
+        onClick={() => retry()}
```
Comment (Tiagoperes) [inline packages/beagle-react/src/components/BeagleError/index.tsx:46]: pode passar só `retry` aqui `onClick={retry}`

File: packages/beagle-react/src/components/BeagleError/index.tsx [hunk 3]
```diff
@@ -1,31 +1,76 @@
...
+      >
+        Retry
+      </StyledRetryButton>
+      {errors.length > 0 ? (
+        <>
+          <StyledShowMoreButton
+            type="button"
+            data-testid="show-more"
+            onClick={() => handleShowMore()}
```
Comment (Tiagoperes) [inline packages/beagle-react/src/components/BeagleError/index.tsx:55]: `onClick={handleShowMore}`

File: packages/beagle-react/src/components/commons.styled.ts [hunk 4]
```diff
@@ -14,11 +14,43 @@
...
-  swampLight:  'rgba(0, 27, 38, 0.8)',
-  swampTransparent:  'rgba(0, 27, 38, 0.1)',
+  swampLight: 'rgba(0, 27, 38, 0.8)',
+  swampTransparent: 'rgba(0, 27, 38, 0.1)',
   athensGray: '#F5F7F9',
   blackTransparent: 'rgba(0, 0, 0, 0.1)',
   darkGray: 'rgba(0, 0, 0, 0.75);',
 }
+
+export const StyledButton = styled.button`
```
Comment (Tiagoperes) [inline packages/beagle-react/src/components/commons.styled.ts:28]: Ao invés de transferir esses estilos de lugar, não seria melhor vc ter usado o componente `<BeagleButton>`?

**Review by Tiagoperes — CHANGES_REQUESTED** [review]

File: packages/beagle-react/package.json (hunk 1 above)
Comment (arthurbleilzup) [reply packages/beagle-react/package.json]: estava indo testar o RN, mas estava na pasta errada, esqueci de remover a dep errada

File: packages/beagle-react/src/components/BeagleError/index.tsx (hunk 2 above)
Comment (arthurbleilzup) [reply packages/beagle-react/src/components/BeagleError/index.tsx]: resolvido

File: packages/beagle-react/src/components/commons.styled.ts (hunk 4 above)
Comment (arthurbleilzup) [reply packages/beagle-react/src/components/commons.styled.ts]: resolvido

**Review by hectorcustodiozup — DISMISSED** [review]

**Review by Tiagoperes — DISMISSED** [review]

**Review by hectorcustodiozup — APPROVED** [review]

**Review by Tiagoperes — APPROVED** [review]

**Pull request merged**